apps/transaction/tests/test_forms/test_transaction_create_form_on_commit.py

```
@@ -0,0 +1,113 @@
+from apps.currency.tests.factories import CurrencyFactory
+from apps.room.tests.factories import RoomFactory
+from apps.transaction.forms.transaction_create_form import TransactionCreateForm
+from apps.transaction.models import ParentTransaction
+
+pytestmark = pytest.mark.django_db
+
+
+class TestTransactionCreateFormOnCommit:
+    """handle_message must be deferred to on_commit, not called inside the atomic block."""
+
+    def _build_form_data(self, room, user, currency, paid_for_users, description="Test transaction"):
```

Ruff (0.15.10)

[warning] Missing return type annotation for private function `_build_form_data`

(ANN202)